lib/python/picongpu/plugins/plot_mpl/emittance_evolution_visualizer.py

```diff
@@ -0,0 +1,164 @@
+        self.cbar = None
+
+    def _create_data_reader(self, run_directory):
+        """
+        Implementation of base class function.
+        """
+        return EmittanceData(run_directory)
+
+    def _create_plt_obj(self, ax):
+        """
+        Implementation of base class function.
+        Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+        """
+        emit, y_slices, all_iterations, dt = self.data
+        np_data = np.zeros(len(all_iterations))
+        for index, ts in enumerate(all_iterations):
+            np_data[index] = emit[ts][0]
+        ps = 1.e12  # for conversion from s to ps
+        # np_data * 1.e6 converts emittance to pi mm mrad
+        self.plt_obj, = ax.plot(all_iterations * dt * ps,
+                                np_data * 1.e6, scalex=True, scaley=True)
+        if self.iteration:
+            self.plt_lin = ax.axvline(self.iteration * dt * ps,
+                                      color='#FF6600')
+
+    def _update_plt_obj(self):
+        """
+        Implementation of base class function.
+        """
+        emit, y_slices, all_iterations, dt = self.data
+        np_data = np.zeros(len(all_iterations))
+        for index, ts in enumerate(all_iterations):
+            np_data[index] = emit[ts][0]
+        if self.plt_lin:
+            self.plt_lin.remove()
+        ps = 1.e12  # for conversion from s to ps
+        # np_data * 1.e6 converts emittance to pi mm mrad
+        self.plt_obj.set_data(all_iterations * dt * ps, np_data * 1.e6)
+        if self.cur_iteration:
+            self.plt_lin = self.ax.axvline(self.cur_iteration * dt * ps,
+                                           color='#FF6600')
+        self.ax.relim()
+        self.ax.autoscale_view(True, True, True)
+
+    def visualize(self, ax=None, **kwargs):
+        """
+        Creates a semilogy plot on the provided axes object for
+        the data of the given iteration using matpotlib.
+
+        Parameters
+        ----------
+        iteration: int
+            the iteration number for which data will be plotted.
+        ax: matplotlib axes object
+            the part of the figure where this plot will be shown.
+        kwargs: dictionary with further keyword arguments, valid are:
+            species: string
+                short name of the particle species, e.g. 'e' for electrons
+                (defined in ``speciesDefinition.param``)
+            iteration: int
+                number of the iteration
+            species_filter: string
+                name of the particle species filter, default is 'all'
+                (defined in ``particleFilters.param``)
+
+        """
+        self.ax = self._ax_or_gca(ax)
+        self.cur_iteration = kwargs.get('iteration')
+        # this already throws error if no species or iteration in kwargs
+        kwargs['iteration'] = None
+        super(Visualizer, self).visualize(ax, **kwargs)
+        species = kwargs.get('species')
```

Comment:
move all the lines below (starting from 100) into the `adjust_plot` function which is called always after drawing the plot data (see the `visualize` function in `base_visualizer.py`)